### ISSUE TITLE
Please add "initial value"

### ISSUE THREAD

**Issue by lucroch** [tameemsafi/typewriterjs#130]
I wanted to use the delete effect in React, but I can't figure a way to have a starting text. 
Option for initial value could be great.

**Comment by ianwensink** [comment]
> You can write your initial text and then use the component.

@salma-py97 but when writing a value inside the container and then initialising it, everything get's removed immediately right? That is in the code of the instructor.

So my use case for this is a bit different from @lucroch. I need to support server side rendering, so the string needs to be in the HTML response. This is all working when adding the text by default inside the wrapper element. On the client, though, the typewriter class will be initialised, and will delete the already written text.

**Comment by lucroch** [comment]
I found a workaround for my needs, although having initial value would still be great.
On ReactJS, I could put a state inside the onInit function and change it at any point and it's read.

**Comment by tini2n** [comment]
Have the same problem
@lucroch can you share your workable solution? 🙂

**Comment by Nikola-Milovic** [comment]
@lucroch  same issue, can you post your solution?

**Comment by lucroch** [comment]
Hey there @tini2n and @Nikola-Milovic, I'm really sorry for the delay on responding this.
I solved this a while ago so I don't exactly remember how I did it, but I like I said, I put a state inside the onInit function that I can change at any point.
`    const [writer, setWriter] = useState<TypewriterClass>()
    const [transition, setTransition] = useState("Start")
    
    useEffect(
        () => {
            if (isExit) {
                setTransition("Final")
                writer?.deleteAll().start()
            }
        }, [isExit]
    )
    
    return(
        <LogoText>
                <p className={transition}>Auth</p>
                <p>
                    <Typewriter
                        options={{cursor: ".", delay:55}}
                        onInit={(typer) => {
                            typer.typeString(text).start();
                            setWriter(typer)
                        }}
                    />
                </p>
        </LogoText>
    )`

The end result is pretty much just like this:
[https://camo.githubusercontent.com/7b8d67f7f7d065d8fddee4e0c5ea2c6d665bdd2d9ade3f6ac21b52c20b2c23f3/68747470733a2f2f692e696d6775722e636f6d2f396d7568624e682e676966](url)

**Comment by rohitsachaan** [comment]
@lucroch it will be really helpful if you or anyone else can share their code, to achieve this?

or

@tameemsafi is there any release planned for this feature #118?

**Comment by HenriqueArtur** [comment]
@rohitsachaan  I used the @lucroch  solution as a reference and worked perfectly for me.

Note that I used a Hook, but you could change it to a variable using "useState"

```TypeScript
import { useEffect, useState } from "react";
import useTitle from "../../hooks/useTitle";
import Typewriter, { TypewriterClass } from 'typewriter-effect';

export default function Title(): JSX.Element {
  const title = useTitle();
  const [writer, setWriter] = useState<TypewriterClass>();

  const animate = (typewriter: any) => {
    typewriter
      .typeString(title)
      .start();
    setWriter(typewriter);
  };

  useEffect(() => {
    if (writer) {
      writer
        .deleteAll(15)
        .typeString(title)
        .start()
      setWriter(writer);
    }
  }, [title]);

  return (
    <h1>
      <Typewriter
        onInit={animate}
        options={{ cursor: ".", delay: 35 }}
      />
    </h1>
  );
};
```

**Comment by mwskwong** [comment]
> @rohitsachaan I used the @lucroch solution as a reference and worked perfectly for me.
> 
> Note that I used a Hook, but you could change it to a variable using "useState"
> 
> ```ts
> import { useEffect, useState } from "react";
> import useTitle from "../../hooks/useTitle";
> import Typewriter, { TypewriterClass } from 'typewriter-effect';
> 
> export default function Title(): JSX.Element {
>   const title = useTitle();
>   const [writer, setWriter] = useState<TypewriterClass>();
> 
>   const animate = (typewriter: any) => {
>     typewriter
>       .typeString(title)
>       .start();
>     setWriter(typewriter);
>   };
> 
>   useEffect(() => {
>     if (writer) {
>       writer
>         .deleteAll(15)
>         .typeString(title)
>         .start()
>       setWriter(writer);
>     }
>   }, [title]);
> 
>   return (
>     <h1>
>       <Typewriter
>         onInit={animate}
>         options={{ cursor: ".", delay: 35 }}
>       />
>     </h1>
>   );
> };
> ```

Too bad, the default value will not be available in SSR

**Comment by viatanas** [comment]
I agree that an initial value attribute is really important and would be a great addition to the package. From a UX perspective, you don't want the user to get on a page, without any initial content on it.

**Comment by CanRau** [comment]
Came here for the same to get rid of the cumulative layout shift on SSR rendered page when the animation first starts

**Comment by ali-garajian** [comment]
@tameemsafi would you accept a PR for this?
